File: Sources/NSBitmapImageRep-Extension.m
#import "NSBitmapImageRep-Extension.h"

@implementation NSBitmapImageRep (Extension)
-(NSData*)representationUsingType:(CFStringRef)type usingDPI:(NSInteger)dpi
{
    NSDictionary<NSString*, id> *prop = @{(__bridge_transfer NSString*)kCGImageDestinationLossyCompressionQuality : @(1.0),
                                          (__bridge_transfer NSString*)kCGImagePropertyDPIWidth : @(dpi),
                                          (__bridge_transfer NSString*)kCGImagePropertyDPIHeight : @(dpi)
                                          };

    NSMutableData *outputData = [NSMutableData data];
    CGImageDestinationRef destination =  CGImageDestinationCreateWithData((CFMutableDataRef)outputData, type, 1, NULL);
    CGImageDestinationAddImage(destination, self.CGImage, (CFDictionaryRef)prop);
    CGImageDestinationFinalize(destination);
    
    CFRelease(destination);
    
    return outputData;
}
@end

File: Sources/NSBitmapImageRep-Extension.swift
import Cocoa

extension NSBitmapImageRep {
    @objc func representation(usingType type: CFString, usingDPI dpi: Int) -> Data? {
        let prop : [String:NSNumber] = [
            kCGImageDestinationLossyCompressionQuality as String: 1.0,
            kCGImagePropertyDPIWidth as String: NSNumber(value: dpi),
            kCGImagePropertyDPIHeight as String: NSNumber(value: dpi)
        ]

        let data = NSMutableData() as CFMutableData
        guard
              let destination = CGImageDestinationCreateWithData(data, type, 1, nil),
              let cgImage = self.cgImage else { return nil }

        CGImageDestinationAddImage(destination, cgImage, prop as CFDictionary)
        CGImageDestinationFinalize(destination)

        return data as Data?
    }
}

